pkg/controller/machine.go

```diff
@@ -312,6 +312,20 @@ func (c *controller) updateMachineState(machine *v1alpha1.Machine) (*v1alpha1.Ma
 	if err != nil {
 		return machine, err
 	}
+
+	if clone.Labels == nil {
```

Comment:
I think we need to take a deepCopy before making the changes here?

Reply:
thanks, I wasn't really sure to deep-copy here. We can certainly deep-copy, do you have any possible reasoning to support it?

Reply:
From what I understand, we would be working from an object from the local cache directly. if we don't deep copy. Deep copying creates a local copy on this method and then acts on it.

Reply:
As discussed over a call, made few changes.
Thanks, this was a good catch..

Reply:
Looks good now.